tests/Microsoft.Identity.Test.Unit.net45/CoreTests/StringExtensionTests.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Test.Common.Core.Helpers;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Test.Unit.CoreTests
+{
+    [TestClass]
+    public class StringExtensionTests
+    {
+        [TestMethod]
+        public void StringReplace()
+        {
+            string original1 = "https://login.microsoftonline.com/{tenant}/oauth2";
+            string original2 = "https://login.microsoftonline.com/{tenant_id}/oauth2";
+            string expected = "https://login.microsoftonline.com/common/oauth2";
```

Comment:
you should have tests with multiple {xxx} values in the template string to ensure it handles multiple insertion as well.